### PR TITLE
Support Monacoin.

### PR DESCRIPTION
**Pull request by cryptcoin-junkey**
Currently the status of this PR is working in progress.
It is for avoiding duplicate works between Monacoiners.
I've not tested this enough yet even though It might work fine.

I'll write the comment after this branch gets stable. Please suspend merging for now.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Remove the last line, when other NBitcoin contributor run the tests, they should run bitcoin by default. Monacoin should stay commented, it is just so people interested to debug monacoin can easily switch it on.

**Comment by cryptcoin-junkey** [comment]
Currently I'm running CI test on VisutalStudio Team Service. I'll revert test code after all issues are fixed.

**Comment by cryptcoin-junkey** [comment]
@NicolasDorier Let me make sure. Are Tests fit to the current API set? At least, sample code in README.md looks obsolete.

**Comment by NicolasDorier** [comment]
Only the `Altcoins=Altcoins` tests need to be ran for testing your altcoin implementation.

**Comment by cryptcoin-junkey** [comment]
At least, 'Altcoins.Litecoin.EnsureRegistered()` will cause a build error like follows.
It is shown in [NBitcoin.Altcoins/README.md](https://github.com/MetacoSA/NBitcoin/blame/d738e5af32d10cd3918c0c65be478431a0808c22/NBitcoin.Altcoins/README.md#L40)

```
NodeBuilderEx.cs(34,4): error CS0120: An object reference is required for the non-static field, method, or property 'NetworkSetBase.EnsureRegistered()' [C:\projects\nbitcoin\NBitcoin.Tests\NBitcoin.Tests.csproj]
```

And I got some assertion failures caused by the value stored in `Network`.

**Comment by cryptcoin-junkey** [comment]
And `Altcoins.Litecoin.Regtest` also (at https://github.com/MetacoSA/NBitcoin/blame/d738e5af32d10cd3918c0c65be478431a0808c22/NBitcoin.Altcoins/README.md#L41 ) also looks obsolete.

**Comment by NicolasDorier** [comment]
Any other issue?

**Comment by NicolasDorier** [comment]
Fix test failing which might gave you issues, please rebase.

File: NBitcoin.Altcoins/Monacoin.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+			{
+				return new MonacoinBlock(new MonacoinBlockHeader());
+			}
+		}
+
+		public class MonacoinBlockHeader : BlockHeader
+		{
+			public override uint256 GetPoWHash()
+			{
+				//TODO: Fix code in this function.
```
Comment (cryptcoin-junkey) [inline NBitcoin.Altcoins/Monacoin.cs]: The last issue for this pull request is caused by here. Monacoin was hard-forked before.

**Comment by NicolasDorier** [comment]
Pushed new nuget package,

Next you might want to integrate into https://github.com/dgarage/NBXplorer 
This is relatively easy compared to this.
Then after we can add support to BTCPay.

**Pull request merged**